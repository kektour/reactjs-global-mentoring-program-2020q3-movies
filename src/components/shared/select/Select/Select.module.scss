.root {
  border: none;
  font-size: 16px;
  color: #fff;
  appearance: none;
  background: transparent;
  background-image: url("data:image/svg+xml;utf8,<svg fill='%23f65261' height='24' viewBox='0 0 24 24' width='24' xmlns='http://www.w3.org/2000/svg'><path d='M7 10l5 5 5-5z'/><path d='M0 0h24v24H0z' fill='none'/></svg>");
  background-repeat: no-repeat;
  background-position-x: 100%;
  background-position-y: 50%;
  padding-right: 24px;

  &:focus {
    outline: none;
  }

  &:hover {
    cursor: pointer;
  }
}

.option {
  color: #000;
}
