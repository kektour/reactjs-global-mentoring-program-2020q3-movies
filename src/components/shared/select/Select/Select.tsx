import React from 'react';
import classnames from 'classnames';
import styles from './Select.module.scss';

export interface Option {
  label: string;
  value: string;
}

type Props = {
  classes?: {
    root?: string;
    option?: string;
  };
  value: string | Array<string>;
  options: Array<Option>;
  multiple?: boolean;
  onChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
};

export const Select: React.FC<Props> = (props) => {
  const { classes = {}, multiple = false, options, value, onChange: handleChange } = props;
  return (
    <select className={classnames(styles.root, classes.root)} value={value} multiple={multiple} onChange={handleChange}>
      <option value="" disabled hidden>
        Choose here
      </option>
      {options.map((o) => (
        <option key={o.value} value={o.value} className={classnames(styles.option, classes.option)}>
          {o.label}
        </option>
      ))}
    </select>
  );
};
