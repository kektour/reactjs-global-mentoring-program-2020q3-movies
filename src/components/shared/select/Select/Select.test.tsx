import React from 'react';
import { render } from '@testing-library/react';
import { Select, Option } from './Select';

describe('Select', () => {
  it('Should render component correctly', () => {
    const options: Array<Option> = [
      {
        label: 'Foo',
        value: 'foo',
      },
      {
        label: 'Bar',
        value: 'bar',
      },
      {
        label: 'Baz',
        value: 'baz',
      },
    ];
    const component = render(
      <Select
        options={options}
        nativeSelectProps={{
          defaultValue: 'foo',
        }}
      />
    );
    expect(component.asFragment()).toMatchSnapshot();
  });
});
