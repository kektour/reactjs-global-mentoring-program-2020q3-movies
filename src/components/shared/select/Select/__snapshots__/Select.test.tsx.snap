// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Should render component correctly 1`] = `
<DocumentFragment>
  <select
    class="root"
  >
    <option
      disabled=""
      hidden=""
      value=""
    >
      Choose here
    </option>
    <option
      class="option"
      selected=""
      value="foo"
    >
      Foo
    </option>
    <option
      class="option"
      value="bar"
    >
      Bar
    </option>
    <option
      class="option"
      value="baz"
    >
      Baz
    </option>
  </select>
</DocumentFragment>
`;
